src/realm/object-store/sync/sync_session.hpp

```diff
@@ -429,6 +437,9 @@ class SyncSession : public std::enable_shared_from_this<SyncSession> {
     // Create active subscription set after PBS -> FLX migration to cover the data.
     void make_active_subscription_set() REQUIRES(!m_state_mutex);
 
+    // Return the subscription_store_base - to be used only for testing
+    std::shared_ptr<sync::SubscriptionStore> get_subscription_store_base() REQUIRES(!m_state_mutex);
```

Comment:
Would it make sense to make `OnlyForTesting` a friend of `SyncSession`? You don't need to expose this then..